### PR TITLE
Integrate Date class with AddCommand

### PR DESCRIPTION
**Pull request by Xilef121**
AddCommand is unable to add company with dates properly.

Edit AddCommandParser to parse date arguments correctly to allow adding with dates.

Let's,
* update the AddCommandParser to check if startDate and endDate prefix is present
* properly show the period of time in the gui
* change resultbox to only show phone and dates if they are present
* enable saving and loading of dates for company

### CODE REVIEW

File: src/test/java/seedu/address/testutil/CompanyBuilder.java [hunk 1]
```diff
@@ -20,19 +21,25 @@ public class CompanyBuilder {
...
+    private Date startDate;
+    private Date endDate;
 
     /**
      * Creates a {@code CompanyBuilder} with the default details.
      */
     public CompanyBuilder() {
         name = new Name(DEFAULT_NAME);
-        phone = new Phone(DEFAULT_PHONE);
+        phone = new Phone("85355255");
```
Comment (jasperng-nus) [inline src/test/java/seedu/address/testutil/CompanyBuilder.java:39]: I think this could be left as DEFAULT_PHONE

**Review by jasperng-nus — COMMENTED** [review]
Other than the test cases, and the default phone number in CompanyBuilder, everything LGTM!

File: src/main/java/seedu/address/model/company/Date.java [hunk 2]
```diff
@@ -30,7 +32,7 @@ public Date(String date) {
      * Constructs an empty {@code Date}
      */
     public Date() {
-        this.date = null;
+        this.date = LocalDate.parse(DEFAULT_DATE_STRING);
```
Comment (cocoanautz) [inline src/main/java/seedu/address/model/company/Date.java:35]: If using the singleton pattern, this.date should point to the DEFAULT_DATE

File: src/main/java/seedu/address/model/company/Date.java [hunk 3]
```diff
@@ -47,7 +49,7 @@ public static boolean isValidDate(String test) {
 
     @Override
     public String toString() {
-        return date.toString();
+        return this.date.toString();
```
Comment (AjayArvind2207) [inline src/main/java/seedu/address/model/company/Date.java:52]: Do you think it will be better to not use "this" since we have not used it elsewhere?

**Review by AjayArvind2207 — COMMENTED** [review]
Looks good on the whole! Just a minor nit related to styling.

File: src/main/java/seedu/address/logic/Messages.java [hunk 4]
```diff
@@ -36,12 +38,21 @@ public static String getErrorMessageForDuplicatePrefixes(Prefix... duplicatePref
...
-                .append(company.getEmail())
-                .append("; Tags: ");
+        builder.append(company.getName());
+        if (!company.getPhone().value.equals(Phone.DEFAULT_NUMBER)) {
+            builder.append("; Phone: ")
+                    .append(company.getPhone());
+        }
+        builder.append("; Email: ")
+                .append(company.getEmail());
+        if (!company.getStartDate().getDate().equals(Date.getDefaultDate())
```
Comment (cocoanautz) [inline src/main/java/seedu/address/logic/Messages.java:48]: The implementation of the Date class should not be exposed here. Handle equality check internally. In lieu of time, this change can be during the next PR.

**Review by cocoanautz — COMMENTED** [review]
LGTM. Just take note of the design improvements.

File: src/test/java/seedu/address/testutil/CompanyBuilder.java (hunk 1 above)
Comment (Xilef121) [reply src/test/java/seedu/address/testutil/CompanyBuilder.java:39]: resolved

File: src/main/java/seedu/address/model/company/Date.java (hunk 2 above)
Comment (Xilef121) [reply src/main/java/seedu/address/model/company/Date.java:35]: done

File: src/main/java/seedu/address/model/company/Date.java (hunk 3 above)
Comment (Xilef121) [reply src/main/java/seedu/address/model/company/Date.java:52]: done

File: src/main/java/seedu/address/model/company/Date.java (hunk 2 above)
Comment (Xilef121) [reply src/main/java/seedu/address/model/company/Date.java:35]: resolved

File: src/main/java/seedu/address/logic/Messages.java (hunk 4 above)
Comment (Xilef121) [reply src/main/java/seedu/address/logic/Messages.java:48]: has been handled to use a method in Date to check if Date is present or not.

**Comment by AjayArvind2207** [comment]
Modifications look good to me! The design change looks appropriate👍

**Review by SherwynNg — COMMENTED** [review]
For the startDate and endDate, what if the endDate is earlier than the startDate? Will it throw an error?

**Pull request merged**